src/Illuminate/Process/PendingProcess.php

```diff
@@ -382,6 +384,8 @@ protected function resolveSynchronousFake(string $command, Closure $fake)
      * @param  callable|null  $output
      * @param  \Closure  $fake
      * @return \Illuminate\Process\FakeInvokedProcess
+     *
+     * @throw \LogicException
```

Comment:
This exception is not needed!

Reply:
At the end of this method ( `resolveAsynchronousFake` ), we have this line. 

```php
 
throw new LogicException('Unsupported asynchronous process fake result provided.');

```

If none of the if conditions is satisfied, this exception is thrown.